tests/disaster-recovery/sc_arbiter/test_mon_osd_failures.py

```diff
@@ -0,0 +1,212 @@
+import random
+import time
+
+import pytest
+import logging
+
+from datetime import datetime, timezone
+
+from ocs_ci.helpers.helpers import modify_deployment_replica_count
+from ocs_ci.helpers.stretchcluster_helper import recover_workload_pods_post_recovery
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.ocs.resources.pod import (
+    get_not_running_pods,
+    wait_for_pods_to_be_in_statuses,
+    get_deployment_name,
+)
+from ocs_ci.ocs.resources.pvc import get_pvc_objs
+from ocs_ci.ocs.resources.stretchcluster import StretchCluster
+from ocs_ci.ocs import constants
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="class")
+def setup_logwriter_workloads(
+    request,
+    setup_logwriter_cephfs_workload_class,
+    setup_logwriter_rbd_workload_class,
+    logreader_workload_class,
+):
+    sc_obj = StretchCluster()
+    # Run the logwriter cephFs workloads
+    (
+        sc_obj.cephfs_logwriter_dep,
+        sc_obj.cephfs_logreader_job,
+    ) = setup_logwriter_cephfs_workload_class(read_duration=0)
+
+    # Generate 5 minutes worth of logs before inducing the netsplit
+    logger.info("Generating 2 mins worth of log")
+    time.sleep(120)
+
+    sc_obj.get_logwriter_reader_pods(label=constants.LOGWRITER_CEPHFS_LABEL)
+    sc_obj.get_logwriter_reader_pods(label=constants.LOGREADER_CEPHFS_LABEL)
+    sc_obj.get_logwriter_reader_pods(
+        label=constants.LOGWRITER_RBD_LABEL, exp_num_replicas=2
+    )
+    logger.info("All the workloads pods are successfully up and running")
+
+    start_time = datetime.now(timezone.utc)
+    sc_obj.get_logfile_map(label=constants.LOGWRITER_CEPHFS_LABEL)
+    sc_obj.get_logfile_map(label=constants.LOGWRITER_RBD_LABEL)
+
+    def finalizer():
+        """
+        Check for data loss, data corruption at the end of the tests
+
+        """
+        end_time = datetime.now(timezone.utc)
+
+        try:
+            sc_obj.get_logwriter_reader_pods(label=constants.LOGWRITER_CEPHFS_LABEL)
+            sc_obj.get_logwriter_reader_pods(
+                label=constants.LOGREADER_CEPHFS_LABEL,
+                statuses=[constants.STATUS_RUNNING, constants.STATUS_COMPLETED],
+            )
+            sc_obj.get_logwriter_reader_pods(
+                label=constants.LOGWRITER_RBD_LABEL, exp_num_replicas=2
+            )
+        except UnexpectedBehaviour:
+
+            logger.info("some pods are not running, so trying the work-around")
```

Comment:
I did not understand this, what workaround are we applying here? do we need to fail the test if something goes wrong with the app workload during the test?

Reply:
In the next line of the code there is a function (recover_workload_pods_post_recovery()) we call, this function will perform the workaround for a known issue with OCP. We dont have to fail the test that's why we execute the workaround to make the pods start running again.